nrcemt/nanomi_optics/engine/optimization.py

```diff
@@ -0,0 +1,68 @@
+import math
+import numpy as np
+import scipy.optimize
+from .lens import Lens
+
+
+def create_optimizable_funcion(
+    mode, lens_i, lens_location, focal_lengths, rays, active
+):
+    def cf_function(x):
+        if abs(x[0]) > 300 or abs(x[0]) < 6:
+            return math.inf
+
```

Comment:
I think this can be achieved less hackily with the `bounds` parameter to the `least_squares` function
https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.least_squares.html
> bounds2-tuple of array_like, optional
Lower and upper bounds on independent variables. Defaults to no bounds. Each array must match the size of x0 or be a scalar, in the latter case a bound will be the same for all variables. Use np.inf with an appropriate sign to disable bounds on all or some variables.

Reply:
also why is `abs` being used here?